diff --git a/target/product/base_system.mk b/target/product/base_system.mk
index 4569bceff9..3f4eb19cc9 100644
--- a/target/product/base_system.mk
+++ b/target/product/base_system.mk
@@ -51,7 +51,7 @@ PRODUCT_PACKAGES += \
     cmd \
     com.android.adbd \
     com.android.conscrypt \
-    com.android.extservices \
+    ExtServices \
     com.android.i18n \
     com.android.ipsec \
     com.android.location.provider \
@@ -59,6 +59,7 @@ PRODUCT_PACKAGES += \
     com.android.media.swcodec \
     com.android.mediaprovider \
     com.android.os.statsd \
+    libincident \
     com.android.permission \
     com.android.resolv \
     com.android.neuralnetworks \
@@ -79,7 +80,6 @@ PRODUCT_PACKAGES += \
     dpm \
     dumpstate \
     dumpsys \
-    DynamicSystemInstallationService \
     e2fsck \
     ExtShared \
     flags_health_check \
@@ -254,11 +254,7 @@ PRODUCT_PACKAGES += \
     svc \
     task_profiles.json \
     tc \
-    telecom \
-    telephony-common \
     tombstoned \
-    traced \
-    traced_probes \
     tune2fs \
     tzdatacheck \
     uiautomator \
@@ -324,7 +320,6 @@ PRODUCT_BOOT_JARS := \
     $(ART_APEX_JARS) \
     framework-minus-apex \
     ext \
-    telephony-common \
     voip-common \
     ims-common
 
diff --git a/target/product/full_base.mk b/target/product/full_base.mk
index ffd3cde11a..bdcd6ce89b 100644
--- a/target/product/full_base.mk
+++ b/target/product/full_base.mk
@@ -48,14 +48,11 @@ PRODUCT_PROPERTY_OVERRIDES := \
     ro.config.notification_sound=pixiedust.ogg
 
 # Put en_US first in the list, so make it default.
-PRODUCT_LOCALES := en_US
+PRODUCT_LOCALES := en_US zh_CN zh_HK zh_TW
 
 # Get some sounds
 $(call inherit-product-if-exists, frameworks/base/data/sounds/AllAudio.mk)
 
-# Get a list of languages.
-$(call inherit-product, $(SRC_TARGET_DIR)/product/languages_full.mk)
-
 # Get everything else from the parent package
 $(call inherit-product, $(SRC_TARGET_DIR)/product/generic_no_telephony.mk)
 
diff --git a/target/product/handheld_system.mk b/target/product/handheld_system.mk
index 608ad02453..0dfabe2611 100644
--- a/target/product/handheld_system.mk
+++ b/target/product/handheld_system.mk
@@ -53,32 +53,24 @@ PRODUCT_PACKAGES += \
     KeyChain \
     librs_jni \
     ManagedProvisioning \
-    MmsService \
     MtpService \
     MusicFX \
     NfcNci \
     PacProcessor \
-    PrintRecommendationService \
     PrintSpooler \
     ProxyHandler \
     screenrecord \
     SecureElement \
     SharedStorageBackup \
     SimAppDialog \
-    Telecom \
-    TelephonyProvider \
-    TeleService \
     Traceur \
     UserDictionaryProvider \
-    VpnDialogs \
-    vr \
 
 
 PRODUCT_SYSTEM_SERVER_APPS += \
     FusedLocation \
     InputDevices \
     KeyChain \
-    Telecom \
 
 PRODUCT_COPY_FILES += \
     frameworks/av/media/libeffects/data/audio_effects.conf:system/etc/audio_effects.conf
